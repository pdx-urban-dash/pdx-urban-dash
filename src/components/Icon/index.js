import React from 'react';
import PropTypes from 'prop-types';
import {
  Target,
  TrendingUp,
  TrendingDown,
  Maximize,
  Minimize,
  GitHub,
} from 'react-feather';

import { colors, iconSizes } from '../../constants';

const Icon = ({ size, type }) => {
  const validSizes = {
    sm: iconSizes.small,
    md: iconSizes.medium,
    lg: iconSizes.large,
    xl: iconSizes.xlarge,
  };

  const mSize = Object.keys(validSizes).includes(size) ? size : 'md';

  switch (type.toLowerCase()) {
    case 'target':
      return <Target size={validSizes[mSize]} />;
    case 'off-target':
      return <Target size={validSizes[mSize]} color={colors.warning} />;
    case 'on-target':
      return <Target size={validSizes[mSize]} color={colors.success} />;
    case 'trending-up':
      return <TrendingUp size={validSizes[mSize]} />;
    case 'trending-down':
      return <TrendingDown size={validSizes[mSize]} />;
    case 'maximize':
      return <Maximize size={validSizes[mSize]} />;
    case 'minimize':
      return <Minimize size={validSizes[mSize]} />;
    case 'github':
      return <GitHub size={validSizes[mSize]} />;
    default:
      throw new Error(`Icon type: ${type} does not exist`);
  }
};

Icon.propTypes = {
  size: PropTypes.string,
  type: PropTypes.string.isRequired,
};

Icon.defaultProps = {
  size: 'md',
};

export default Icon;
