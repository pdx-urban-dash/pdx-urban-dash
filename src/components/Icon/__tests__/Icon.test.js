import React from 'react';
import { mount } from 'enzyme';

import Icon from '../index';

describe('<Icon />', () => {
  test('Icon renders target', () => {
    const wrapper = mount(<Icon type="target" />);
    expect(wrapper).toMatchSnapshot();
  });
});
