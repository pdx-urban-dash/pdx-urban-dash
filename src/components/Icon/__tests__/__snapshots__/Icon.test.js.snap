// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Icon renders target 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Icon
    size="md"
    type="target"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "getWrappingComponentRenderer": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "size": "md",
      "type": "target",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "color": "currentColor",
        "size": 36,
      },
      "ref": null,
      "rendered": Object {
        "instance": <svg
          fill="none"
          height="36"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="36"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <circle
            cx="12"
            cy="12"
            r="6"
          />
          <circle
            cx="12"
            cy="12"
            r="2"
          />
        </svg>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <circle
              cx="12"
              cy="12"
              r="10"
            />,
            <circle
              cx="12"
              cy="12"
              r="6"
            />,
            <circle
              cx="12"
              cy="12"
              r="2"
            />,
          ],
          "fill": "none",
          "height": 36,
          "stroke": "currentColor",
          "strokeLinecap": "round",
          "strokeLinejoin": "round",
          "strokeWidth": "2",
          "viewBox": "0 0 24 24",
          "width": 36,
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <circle
              cx="12"
              cy="12"
              r="10"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cx": "12",
              "cy": "12",
              "r": "10",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "circle",
          },
          Object {
            "instance": <circle
              cx="12"
              cy="12"
              r="6"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cx": "12",
              "cy": "12",
              "r": "6",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "circle",
          },
          Object {
            "instance": <circle
              cx="12"
              cy="12"
              r="2"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cx": "12",
              "cy": "12",
              "r": "2",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "circle",
          },
        ],
        "type": "svg",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "size": "md",
        "type": "target",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "currentColor",
          "size": 36,
        },
        "ref": null,
        "rendered": Object {
          "instance": <svg
            fill="none"
            height="36"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="36"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <circle
              cx="12"
              cy="12"
              r="6"
            />
            <circle
              cx="12"
              cy="12"
              r="2"
            />
          </svg>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <circle
                cx="12"
                cy="12"
                r="10"
              />,
              <circle
                cx="12"
                cy="12"
                r="6"
              />,
              <circle
                cx="12"
                cy="12"
                r="2"
              />,
            ],
            "fill": "none",
            "height": 36,
            "stroke": "currentColor",
            "strokeLinecap": "round",
            "strokeLinejoin": "round",
            "strokeWidth": "2",
            "viewBox": "0 0 24 24",
            "width": 36,
            "xmlns": "http://www.w3.org/2000/svg",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <circle
                cx="12"
                cy="12"
                r="10"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "cx": "12",
                "cy": "12",
                "r": "10",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "circle",
            },
            Object {
              "instance": <circle
                cx="12"
                cy="12"
                r="6"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "cx": "12",
                "cy": "12",
                "r": "6",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "circle",
            },
            Object {
              "instance": <circle
                cx="12"
                cy="12"
                r="2"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "cx": "12",
                "cy": "12",
                "r": "2",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "circle",
            },
          ],
          "type": "svg",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
